packages/apollo/src/commands/service/list.ts

```diff
@@ -0,0 +1,173 @@
+import { flags } from "@oclif/command";
+import { ProjectCommand } from "../../Command";
+import { GraphQLSchema } from "graphql";
+import sortBy from "lodash.sortby";
+import { table } from "heroku-cli-util";
+import moment from "moment";
+import { ApolloConfig, isServiceProject } from "apollo-language-server";
+import chalk from "chalk";
+import {
+  ListServices_service_implementingServices,
+  ListServices_service_implementingServices_FederatedImplementingServices_services
+} from "apollo-language-server/lib/graphqlTypes";
+
+interface TasksOutput {
+  config: ApolloConfig;
+  implementingServices: ListServices_service_implementingServices | null;
+}
+
+const formatImplementingService = (
+  implementingService: ListServices_service_implementingServices_FederatedImplementingServices_services
+) => {
+  return {
+    name: implementingService.name,
+    url: implementingService.url || "",
+    updatedAt: `${moment(implementingService.updatedAt).format(
+      "D MMMM YYYY"
+    )} (${moment(implementingService.updatedAt).fromNow()})`
+  };
+};
+
+function formatHumanReadable({
+  implementingServices,
+  graphName,
+  frontendUrl
+}: {
+  implementingServices: ListServices_service_implementingServices | null;
+  graphName: string | undefined;
+  frontendUrl: string | undefined;
+}): string {
+  let result = "";
+  if (
+    !implementingServices ||
+    implementingServices.__typename === "NonFederatedImplementingService"
```

Comment:
at the moment, there is no way to get here, but since `implementingServices` can be either a federated or non federated implementing service at the moment (union of types) I decided to double check. Do you think it's not necessary?

Reply:
I think TS requires this check based on the generated types and I fully support this! It may read a little more clearly to refactor the logic, but this is just a nit.

```js
if (!(implementingServices && implementingServices.__typename === "FederatedImplementingService") {
  ...
}
```